middleman-core/lib/middleman-core/preview_server.rb

```diff
@@ -205,9 +207,19 @@ def setup_webrick(is_logging)
         end
 
         begin
+          tries ||= 4
+          tried_ports ||= []
+
           ::WEBrick::HTTPServer.new(http_opts)
         rescue Errno::EADDRINUSE
-          logger.error "== Port #{port} is unavailable. Either close the instance of Middleman already running on #{port} or start this Middleman on a new port with: --port=#{port.to_i + 1}"
+          unless (tries -= 1).zero?
+            logger.error "== Port #{port} is unavailable. Trying #{port + 1}."
+            tried_ports << (http_opts[:Port] = @port += 1)
+
+            retry
+          end
+
+          logger.error "== Port(s) #{tried_ports.join(', ')} is/are unavailable. Either close the instance of Middleman already running on #{tried_ports.join(', ')} or start this Middleman on a new port with: --port=#{port + 1}"
```

Comment:
You can use `tried_ports.to_sentence` to format it grammatically, and it'd be good to store it in a local to avoid repeating the expression.